### DISCUSSION TITLE
Why result different between groupby and count?

### DISCUSSION THREAD

**Discussion by tommyhj217** [vaexio/vaex#1989] [Q&A] [answered]
Hi 

Thanks for make surprise lib.

I tested count() function and compare with group by result.

I used below data.
taxi_path = 's3://vaex/taxi/yellow_taxi_2012_zones.hdf5?anon=true'

![image](https://user-images.githubusercontent.com/86287388/160237550-5341a3cf-5533-42f0-a9c8-f6e06d1b1ff0.png)

![image](https://user-images.githubusercontent.com/86287388/160237631-cda7eeb7-4971-48a7-b844-42ccdbfa63d7.png)

The other values are all the same, but the last value of each of the last values is output as zero.
Is there a way to solve this problem?

Thanks
Hyunjun

**Comment by maartenbreddels** [comment, marked as answer]
This is expected, since binby will use regular bins between a min and max, with half open intervals, e.g. [0, 1), ... [5, 6).
Then the bins is calculated as int(data / (max - min) * shape) (e.g. `6 => int(6 / 6 * 7) = int(7.0) = 7`, which falls out of the last bin, and `5 => int(5/7*7) = int(5/6*8) = int(5.8333) = 5`. So no data falls in bin 6.

If you pass in the limits, e.g.:
```
df.count('*', binby=['pickup_hour', 'pickup_day'], shape=[24,7], limits=([-0.5, 23.5], [-0.5, 6.5]))
```

You get:

<img width="1118" alt="image" src="https://user-images.githubusercontent.com/1765949/162726446-49e9081e-08af-45fb-9043-b1ae49485607.png">


Does that make sense?

**Reply by tommyhj217** [reply]
Thanks for reply my question.
It seems to be one of the best tools available on a single computer.
